mitodl/teachersportal
# Integrated stripe

#### What's this PR do?
- Adds cart reducer and actions
- Adds code to process Stripe payments in the Python code
- Adds code for Stripe Checkout in the Javascript side
#### Where should the reviewer start?

`portal/views/checkout_api.py` and `static/js/index_page.js`
#### How should this be manually tested?
- Create a Stripe test account
- Add `STRIPE_PUBLISHABLE_KEY` and `STRIPE_SECRET_KEY` with appropriate values to `.env`
- On a course detail page go to the Buy tab
- Click 'Add to cart'. This will add the first module from the course to the cart (currently not serialized anywhere)
- Click 'Pay with card'. Use the card number 4242 4242 4242 4242 with an expiration date set in the future and any three digit code. Remember the amount that is being charged. Also verify that you typed in a valid address to get to this point.
- On success check your network tab for a 200 response code
- Go into the stripe console and verify that successful payment exists in their dashboard, with the same amount as you saw in the earlier dialog.
- Run `./manage.py shell` and verify that the `Order` and `OrderLine` models contain the information about the order that was made. There should be one `OrderLine` for that order and it should connect to the `Product` which is the first module on the course dialog page. 
#### What are the relevant tickets?

Fixes #123
#### What gif best describes this PR or how it makes you feel?

![](https://media2.giphy.com/media/EGn6QuuvGO9BS/200.gif)
